fem/gridfunc.cpp

```diff
@@ -30,6 +30,8 @@ using namespace std;
 GridFunction::GridFunction(Mesh *m, std::istream &input)
    : Vector()
 {
+   UseDevice();
```

Comment:
Well, I tried to explain its purpose in the doxygen documentation. Do you have any suggestions how to improve the documentation?

Reply:
It's difficult because I'm still not sure I fully understand what it does. My understanding is that `UseDevice()` is actually not even used by most kernels (`MFEM_FORALL`), so we have this complex layer of code that is only used in the `Vector` kernels (`MFEM_FORALLIF`), but nowhere else. But even where is `UseDevice()` is used, it's used through `GetExecFlag()` and then the developer has to do a logical "or" between all the variables in the kernel before giving the result of this "or" as the first argument of `MFEM_FORALLIF`? Which makes me wonder, why all this complexity? Why are you so sure that it's so important to not do certain `Vector` tasks on GPU? Is it based on some performance analysis or pure speculation?
If that is the case this clearly seems like early optimization to me, or I'm missing the point because this desoen't really seem to bring any important feature for now.
This drives me to the question, what is your heuristic to decide if an operation should be done on GPU or CPU? Does the size of the `Vector` matters? If yes wouldn't it be sufficient to check the size of the `Vector`s before running the kernel (`bool use_dev = vec.Size() > GPU_SIZE` type of logic)?